simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java

```diff
@@ -47,41 +48,42 @@ protected ByteArrayOutputStream initialValue()
      * Handles Metrics collections from the given registry.
      */
     static class HTTPMetricHandler implements HttpHandler {
-        private CollectorRegistry registry;
+        private final CollectorRegistry registry;
         private final LocalByteArray response = new LocalByteArray();
         private final static String HEALTHY_RESPONSE = "Exporter is Healthy.";
+        private final Charset UTF_8 = Charset.forName("UTF-8");
 
         HTTPMetricHandler(CollectorRegistry registry) {
           this.registry = registry;
         }
 
-
+        @Override
         public void handle(HttpExchange t) throws IOException {
             String query = t.getRequestURI().getRawQuery();
 
             String contextPath = t.getHttpContext().getPath();
             ByteArrayOutputStream response = this.response.get();
             response.reset();
-            OutputStreamWriter osw = new OutputStreamWriter(response, Charset.forName("UTF-8"));
+            OutputStreamWriter osw = new OutputStreamWriter(response, UTF_8);
             if ("/-/healthy".equals(contextPath)) {
                 osw.write(HEALTHY_RESPONSE);
             } else {
                 TextFormat.write004(osw,
                         registry.filteredMetricFamilySamples(parseQuery(query)));
             }
 
-            osw.flush();
             osw.close();
-            response.flush();
```

Comment:
Is it safe to remove all of these?

Reply:
1. `flush` call before `close` is useless. According to javadoc - `close` should call `flush` first.
2. `ByteArrayOutputStream.flush` and `ByteArrayOutputStream.close` are no-op. So it's safe to remove them.